control-plane/csi-driver/src/bin/node/filesystem_ops.rs

```diff
@@ -0,0 +1,332 @@
+        let binary = format!("mkfs.{}", "xfs");
+        let output = Command::new(&binary)
+            .arg(device)
+            .output()
+            .await
+            .map_err(|error| format!("failed to execute {binary}: {error}"))?;
+        ack_command_output(output, binary)
+    }
+
+    fn mount_flags(&self, mount_flags: Vec<String>) -> Vec<String> {
+        let mut mount_flags = mount_flags;
+        if !mount_flags.contains(&XFS_NO_UUID_FLAG.to_string()) {
+            mount_flags.push(XFS_NO_UUID_FLAG.to_string())
+        }
+        mount_flags
+    }
+
+    fn unmount_on_fs_id_diff(
+        &self,
+        device_path: &str,
+        fs_staging_path: &str,
+        volume_uuid: &Uuid,
+    ) -> Result<(), Error> {
+        mount::unmount_on_fs_id_diff(device_path, fs_staging_path, volume_uuid)
+    }
+
+    /// Xfs filesystem needs an unmount to clear the log, so that the parameters can be changed.
+    /// Mount the filesystem to a defined path and then unmount it.
+    async fn repair(
+        &self,
+        device: &str,
+        staging_path: &str,
+        options: &[String],
+        volume_uuid: &Uuid,
+    ) -> Result<(), Error> {
+        mount::filesystem_mount(device, staging_path, &FileSystem(Fs::Xfs), options).map_err(|error| {
+            format!(
+                "(xfs repairing) Failed to mount device {device} onto {staging_path} for {volume_uuid} : {error}",
+            )
+        })?;
+        mount::filesystem_unmount(staging_path).map_err(|error| {
+            format!(
+                "(xfs repairing) Failed to unmount device {device} from {staging_path} for {volume_uuid} : {error}",
+            )
+        })
+    }
+
+    async fn set_uuid(&self, device: &str, volume_uuid: &Uuid) -> Result<(), Error> {
+        if let Ok(probed_uuid) = FileSystem::property(device, "UUID") {
+            if probed_uuid == volume_uuid.to_string() {
+                return Ok(());
+            }
+        }
+
+        let binary = "xfs_admin".to_string();
+        let output = Command::new(&binary)
+            .arg("-U")
+            .arg(volume_uuid.to_string())
+            .arg(device)
+            .output()
+            .await
+            .map_err(|error| format!("failed to execute {binary}: {error}"))?;
+
+        if !output.status.success() {
+            return Err(format!(
+                "{} command failed: {}",
+                binary,
+                String::from_utf8(output.stderr).unwrap()
+            ));
+        }
+
+        let probe_uuid = FileSystem::property(device, "UUID")
+            .map_err(|error| format!("Failed to get UUID of device {device}: {error}"))?;
+
+        if volume_uuid.to_string() != probe_uuid {
+            return Err(format!(
+                "failed to set filesystem uuid using : {binary}, {}",
+                String::from_utf8(output.stdout).unwrap_or_default()
+            ));
+        }
+        Ok(())
+    }
+
+    async fn set_uuid_with_repair(
```

Comment:
Does this need specific or can we have a default impl?

Reply:
Default sounds good.